inspire_dojson/hep/model.py

```diff
@@ -62,16 +62,17 @@ def convert_publication_infos(record, blob):
 
 def move_incomplete_publication_infos(record, blob):
     def _keys_with_truthy_values(d):
-        return [k for k, v in d.items() if v]
+        return set(k for k, v in d.items() if v)
 
     if not record.get('publication_info'):
         return record
 
     for publication_info in record['publication_info']:
-        if _keys_with_truthy_values(publication_info) == ['journal_title']:
+        if _keys_with_truthy_values(publication_info).issubset(set(['journal_record', 'journal_title'])):
             public_note = {'value': u'Submitted to {}'.format(publication_info['journal_title'])}
             record.setdefault('public_notes', []).append(public_note)
-            del publication_info['journal_title']
+            for key in _keys_with_truthy_values(publication_info):
+                del publication_info[key]
```

Comment:
why not delete the whole `publication_info`?

Reply:
I have a feeling that `del`ing while iterating will make Python 3 angry. Let's see what Travis says.

Reply:
yes, you need to copy the list.